io.openems.edge.bridge.modbus/src/io/openems/edge/bridge/modbus/BridgeModbusTcpImpl.java

```diff
@@ -72,6 +86,18 @@ private void modified(ComponentContext context, ConfigTcp config) throws Unknown
 	private void applyConfig(ConfigTcp config) {
 		this.setIpAddress(InetAddressUtils.parseOrNull(config.ip()));
 		this.port = config.port();
+        int coreCycleTime = 1000;
+        try {
+			// the default cycleTime is not persisted and won't be returned, only modified cycleTime will!
+			coreCycleTime = (int) (Integer) this.cm.getConfiguration("Core.Cycle").getProperties().get("cycleTime");
```

Comment:
[Cycle](https://github.com/OpenEMS/openems/blob/develop/io.openems.edge.common/src/io/openems/edge/common/cycle/Cycle.java#L120-L125) has a method `getCycleTime()`. Is there a reason you don't use it directly?

Reply:
Ah sorry. Just realized, that you asked exactly this question here: https://community.openems.io/t/bridgemodbustcpimpl-doesnt-work-when-cycle-is-referenced/2530/3